homeassistant/components/southern_company/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for Southern Company integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from southern_company_api.parser import (
+    CantReachSouthernCompany,
+    InvalidLogin,
+    SouthernCompanyAPI,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    hub = SouthernCompanyAPI(data["username"], data["password"])
+
+    try:
+        await hub.authenticate()
+    except InvalidLogin as err:
+        raise InvalidAuth from err
+    except CantReachSouthernCompany as err:
+        raise CannotConnect from err
+
+    # Return info that you want to store in the config entry.
+    return {"title": "Southern Company"}
```

Comment:
This method isn't needed in this case, all it does it wrapping everything into something else... 

Put it directly in line in the user step instead, which means you can get rid of the `InvalidAuth` and `CannotConnect` error classes as well.